application/forms/EventRuleConfigForm.php

```diff
@@ -0,0 +1,654 @@
+
+            $escalation = new Escalation(
+                $escalationCondition,
+                $escalationRecipient,
+                $removeEscalationButtons[$prefixMap] ?? null
+            );
+
+            if ($zeroConditionEscalation === $prefixMap && $escalation->addConditionHasBeenPressed()) {
+                $this->hasZeroConditionEscalation = false;
+                $zeroConditionEscalation = null;
+            } elseif ($escalation->lastConditionHasBeenRemoved()) {
+                $this->hasZeroConditionEscalation = true;
+                $zeroConditionEscalation = $prefixMap;
+            }
+
+            $escalations[] = $escalation;
+        }
+
+        $this->getElement('zero-condition-escalation')->setValue($zeroConditionEscalation);
+
+        $this->addHtml(
+            (new HtmlElement('ul', Attributes::create(['class' => 'filter-wrapper'])))
+                ->addHtml(
+                    new HtmlElement('li', null, (new FlowLine())->getRightArrow()),
+                    new HtmlElement('li', null, $configFilter),
+                    new HtmlElement('li', null, (new FlowLine())->getHorizontalLine())
+                )
+        );
+
+        $this->add(new Escalations($escalations, $addEscalationButton));
+    }
+
+    /**
+     * Handle addition of escalations
+     *
+     * @return void
+     */
+    protected function handleAdd(): void
+    {
+        $pressedButton = $this->getPressedSubmitElement();
+
+        if ($pressedButton && $pressedButton->getName() === 'add-escalation') {
+            $this->clearPopulatedValue('prefixes-map');
+            $prefixesMapString = $this->getValue('prefixes-map', '');
+            $prefixesMap = explode(',', $prefixesMapString);
+            $escalationFakePos = bin2hex(random_bytes(4));
+            $prefixesMap[] = $escalationFakePos;
+            $this->getElement('prefixes-map')
+                ->setValue(implode(',', $prefixesMap));
+
+            if ($this->getValue('zero-condition-escalation') === null) {
+                $this->getElement('zero-condition-escalation')
+                    ->setValue($escalationFakePos);
+            }
+        }
+    }
+
+    public function populate($values): self
+    {
+        if (! isset($values['rule_escalation'])) {
+            return parent::populate($values);
+        }
+
+        $formValues = [];
+        $formValues['prefixes-map'] = $this->getPrefixesMap(count($values['rule_escalation']));
+
+        foreach ($values['rule_escalation'] as $position => $escalation) {
+            $conditions = explode('|', $escalation['condition'] ?? '');
+            $conditionFormValues = [];
+            $conditionFormValues['condition-count'] = count($conditions);
+            $conditionFormValues['id'] = $escalation['id'] ?? bin2hex(random_bytes(4));
+            foreach ($conditions as $key => $condition) {
+                if ($condition === '' && ! isset($formValues['zero-condition-escalation'])) {
```

Comment:
Please use `empty()` instead.

Reply:
I think that a strict check for empty string is completely alright.